subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -542,7 +561,11 @@ static int get_tx_buffer(const struct device *instance, void *token,
 
 	payload = rpmsg_get_tx_payload_buffer(&rpmsg_ept->ep, size, K_TIMEOUT_EQ(wait, K_FOREVER));
```

Comment:
Do we need here `K_TIMEOUT_EQ(wait, K_FOREVER)`? You are checking `wait` parameter in the line 544: `if (!K_TIMEOUT_EQ(wait, K_NO_WAIT)) {`.